plaid/src/bin/plaid.rs

```diff
@@ -227,47 +228,45 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 
+                                let (response_send, response_recv) = tokio::sync::oneshot::channel();
+
                                 // Construct a message to send to the rule
                                 let message = Message {
                                     type_: name.to_string(),
                                     data: String::new().into_bytes(),
                                     accessory_data: query.into_iter().map(|(k, v)| (k, v.into_bytes())).collect(),
                                     source,
                                     logbacks_allowed,
+                                    response_sender: Some(response_send),
+                                    module: Some(rule.clone()),
                                 };
-                                // We need to spawn blocking here because the executor is not async
-                                // and it will need to block inside the host API functions to call other async
-                                // functions on a different runtime.
-                                //
-                                // As far as I can tell there is no way around this since WASM host functions
-                                // cannot be async. I am afraid this prevents a denial of service risk because of
-                                // the way this works but I don't see a way around it right now.
-                                let rule = rule.clone();
-                                let response = tokio::task::spawn_blocking(move || {
-                                    executor.immediate_execute(message, rule) 
-                                }).await;
-
-                                match response {
-                                    Ok(Ok(Some(response)))=> {
+
+                                // Put the message into the standard message queue
+                                if let Err(e) = log_sender.try_send(message) {
+                                    match e {
+                                        TrySendError::Full(_) => error!("Queue Full! [{}] log dropped!", webhook_configuration.log_type),
+                                        // TODO: Have this actually cause Plaid to exit
+                                        TrySendError::Disconnected(_) => panic!("The execution system is no longer accepting messages. Nothing can continue."),
+                                    }
+                                }
+
+                                match response_recv.await {
+                                    Ok(Some(response))=> {
                                         if update {
                                             info!("Updating cache for get request to: {webhook}");
                                             let mut cache = get_cache.write().await;
                                             cache.insert(webhook.clone(), (current_time, response.clone()));
```

Comment:
I would maybe get a new current time before inserting to the cache here. It's possible that enough time passes here such that the cache is already invalid

Reply:
That's a good callout